ash/src/vk/prelude.rs

```diff
@@ -56,3 +56,7 @@ impl From<vk::Extent2D> for vk::Rect2D {
         }
     }
 }
+
+pub trait HasStructureType {
+    const STRUCTURE_TYPE: vk::StructureType;
+}
```

Comment:
This is made into a trait with the assumption of adding some default `fn`s here that help with iterating `p_next`. I've not really filled in what to do here, and had some thoughts about also having certain getter functions in the form of `fn get_xxx<T: HasStructureType>(...) -> Option<T>` but nothing fleshed/figured out yet.